Week2/ex2.py

```diff
@@ -0,0 +1,21 @@
+import functools
+
+class InvalidTypeError(TypeError):
```

Comment:
It's bad practice to leave a class not implemented.
Search the web to understand how to define a custom error in a good way.